common-rest/src/main/java/org/uniprot/api/common/repository/stream/rdf/RdfStreamer.java

```diff
@@ -47,6 +47,44 @@ public Stream<String> stream(Stream<String> entryIds, String dataType, String fo
                                                 "Finished streaming over search results and fetching from RDF server."));
 
         // prepend rdf prolog then rdf data and then append closing rdf tag
+        return getFullResponseAsStream(dataType, format, rdfStringStream);
+    }
+
+    public Stream<String> stream(
+            Stream<String> entryIds, String dataType, String format, IntConsumer consumer) {
+        BatchRdfXmlStoreIterable batchRDFXMLStoreIterable =
+                getBatchRdfXmlStoreIterable(entryIds, dataType, format);
+
+        Stream<String> rdfStringStream =
+                StreamSupport.stream(batchRDFXMLStoreIterable.spliterator(), false)
+                        .flatMap(Collection::stream)
+                        .map(
+                                response -> {
+                                    consumer.accept(
+                                            rdfEntryCountProvider.getEntryCount(
+                                                    response, dataType, format));
+                                    return response;
```

Comment:
The name response gives an impression that it is full response but it is just a line of the big response. 
line would be more clearer for the reader.